app/src/main/java/org/wildaid/ofish/ui/reportdetail/ReportDetailFragment.kt

```diff
@@ -39,6 +39,7 @@ import org.wildaid.ofish.util.getViewModelFactory
 import org.wildaid.ofish.util.setVisible
 
 const val KEY_REPORT_ID = "report_id"
+const val KEY_SHOULD_SHOW_BUTTON = "should_show_button"
```

Comment:
Please rename this variable like   =  BOARD_VESSEL_ALLOWED

By default all functional should be enabled.. And only withn additional settings we can hide / disable something